src/entities/threshold.rs

```diff
@@ -161,10 +161,9 @@ where
                     .ok_or_else(|| LocalError::new("id={id:?} is missing in the share_ids"))?;
                 Ok((id.clone(), *share_id))
             })
-            .collect::<Result<BTreeMap<_, _>, LocalError>>()?;
+            .collect::<Result<BTreeMap<I, ShareId<P>>, LocalError>>()?;
```

Comment:
This is just to help rust-analyzer resolve the types in the IDE (and perhaps it's helping the casual reader too?).